### ISSUE TITLE
scores mask

### ISSUE THREAD

**Issue by 49879682-dot** [pi-tau/transformer#2]
Hello, I have a question, I think the attention MUST NOT affected by any `<PAD>s`

but, in this annotation, the output of the attention is composed of the weighted values of the `<PAD>s`.

In `MultiHeadAttention::forward`

```python
        if mask is not None: # mask attn logits by setting them to a large negative value
            attn.masked_fill_(~mask, -1e9)
```

the `masked_fill_` change the `attn` (here `attn` is the **scores**)  to:

```
****************####
****************####
....
****************####
-1e9 -1e9 ...-1e9 -1e9
-1e9 -1e9 ...-1e9 -1e9
...
-1e9 -1e9 ...-1e9 -1e9
```

where the `*` and `#` mean the contents unchanged in `attn` when apply `masked_fill_`.

And, the `values` parameter of `MultiHeadAttention::forward` has the form:

```
d_model vector of token1
d_model vector of token2
d_model vector of token...
d_model vector of tokenN
d_model vector of  <PAD>
d_model vector of  <PAD>
...
d_model vector of  <PAD>
```

when `z = torch.matmul(attn, v)` the first N row in the result `z` will have a weighted sum of the `<PAD>s`  because `#` is NOT setted to `-1e9`

the result is not consist with this picture:
![](https://pi-tau.github.io/transformer/mask.png)

And I didn't understand this explanation:

>Note that our attention layer expects the mask to be of shape BxTxT . Simply broadcasting would not produce the exact same mask that we described earlier. However, it achieves the same effect since we don’t really care what the padded elements are attending.

Can you help me whether I misunderstand the code or it was designed to be？

Thanks！
